pkg/k8s/watchers/pod.go

```diff
@@ -685,16 +685,30 @@ func (k *K8sWatcher) updatePodHostData(pod *slim_corev1.Pod, podIPs []string) er
 		// Initial mapping of podIP <-> hostIP <-> identity. The mapping is
 		// later updated once the allocator has determined the real identity.
 		// If the endpoint remains unmanaged, the identity remains untouched.
-		selfOwned, namedPortsChanged := ipcache.IPIdentityCache.Upsert(podIP, hostIP, hostKey, k8sMeta, ipcache.Identity{
+		namedPortsChanged, err := ipcache.IPIdentityCache.Upsert(podIP, hostIP, hostKey, k8sMeta, ipcache.Identity{
 			ID:     identity.ReservedIdentityUnmanaged,
 			Source: source.Kubernetes,
 		})
 		// This happens at most once due to k8sMeta being the same for all podIPs in this loop
 		if namedPortsChanged {
 			k.policyManager.TriggerPolicyUpdates(true, "Named ports added or updated")
 		}
-		if !selfOwned {
-			errs = append(errs, fmt.Sprintf("ipcache entry for podIP %s owned by kvstore or agent", podIP))
+		if err != nil {
+			// It is expected to receive error overwrites where the existing
+			// source is the KVStore, this can happen as KVStore event
+			// propagation can usually be faster than k8s event propagation.
+			// It is also expected to receive an error overwrite where the
+			// existing source is local since cilium-agent receives events for
+			// local pods.
+			if !errors.Is(err, &ipcache.ErrOverwrite{
+				ExistingSrc: source.KVStore,
+				NewSrc:      source.Kubernetes,
+			}) || !errors.Is(err, &ipcache.ErrOverwrite{
+				ExistingSrc: source.Local,
+				NewSrc:      source.Kubernetes,
+			}) {
+				errs = append(errs, fmt.Sprintf("ipcache entry for podIP %s: %s", podIP, err))
```

Comment:
Nit:

```suggestion
				errs = append(errs, fmt.Sprintf("ipcache entry for podIP %s: %w", podIP, err))
```

Reply:
Never mind, I didn't realize `errs` is `[]string` and thus `fmt.Sprintf` instead of `fmt.Errorf` is used.